### PR TITLE
Rhel8 alpha branch blivet3

### PR DESCRIPTION
**Pull request by rvykydal**
Porting these merges from master:
- https://github.com/rhinstaller/anaconda/commit/80a02e2031671aab6a00af72adfeebf3c500794d
- https://github.com/rhinstaller/anaconda/commit/6a19c0d3acb50f96518b847d6456c9ab3c1890bf with https://github.com/rhinstaller/anaconda/commit/291d1e6758d458b0481f4317795a12c900759ede and https://github.com/rhinstaller/anaconda/commit/839bfdde12461f6ad904391c8ddd467c86cfaefe excluded

### CODE REVIEW

**Comment by rvykydal** [comment]
I've checked that all the failing test seem to be caused by testing with blivet 2

**Comment by rvykydal** [comment]
Added rhbz refernces

File: pyanaconda/storage/autopart.py [hunk 1]
```diff
@@ -0,0 +1,522 @@
...
+                disks.append(disk)
+                break
+
+            part = part.nextPartition()
+
+    return disks
+
+
+def _schedule_implicit_partitions(storage, disks):
+    """ Schedule creation of a lvm/btrfs member partitions for autopart.
```
Comment (jkonecny12) [inline pyanaconda/storage/autopart.py]: Remove `btrfs`. It's dead on RHEL8.

File: pyanaconda/storage/autopart.py [hunk 2]
```diff
@@ -0,0 +1,522 @@
...
+            fmt_args = {"passphrase": luks_data.encryption_passphrase,
+                        "cipher": storage.encryption_cipher,
+                        "escrow_cert": storage.autopart_escrow_cert,
+                        "add_backup_passphrase": storage.autopart_add_backup_passphrase,
+                        "min_luks_entropy": luks_data.min_entropy}
+        else:
+            if storage.autopart_type in (AUTOPART_TYPE_LVM, AUTOPART_TYPE_LVM_THINP):
+                fmt_type = "lvmpv"
+            else:
+                fmt_type = "btrfs"
```
Comment (jkonecny12) [inline pyanaconda/storage/autopart.py]: The same here and so on.

File: pyanaconda/storage/osinstall.py [hunk 3]
```diff
@@ -0,0 +1,2197 @@
...
+import time
+import parted
+import shutil
+
+import gi
+gi.require_version("BlockDev", "2.0")
+
+from gi.repository import BlockDev as blockdev
+
+from pykickstart.constants import AUTOPART_TYPE_LVM, CLEARPART_TYPE_NONE, CLEARPART_TYPE_LINUX, CLEARPART_TYPE_ALL, CLEARPART_TYPE_LIST
```
Comment (jkonecny12) [inline pyanaconda/storage/osinstall.py]: Line is too long please wrap it.

File: pyanaconda/storage/osinstall.py [hunk 4]
```diff
@@ -0,0 +1,2197 @@
...
+
+from blivet import arch, util, udev
+from blivet.blivet import Blivet
+from blivet.storage_log import log_exception_info
+from blivet.devices import FileDevice, NFSDevice, NoDevice, OpticalDevice, NetworkStorageDevice, DirectoryDevice, MDRaidArrayDevice, PartitionDevice, BTRFSSubVolumeDevice, TmpFSDevice, LVMLogicalVolumeDevice, LVMVolumeGroupDevice, BTRFSDevice
+from blivet.errors import FSTabTypeMismatchError, UnrecognizedFSTabEntryError, StorageError, FSResizeError, FormatResizeError, UnknownSourceDeviceError
+from blivet.formats import get_device_format_class
```
Comment (jkonecny12) [inline pyanaconda/storage/osinstall.py]: Lines are too long please wrap it.

File: pyanaconda/storage/osinstall.py [hunk 5]
```diff
@@ -0,0 +1,2197 @@
...
+        if not mounts and not swaps:
+            # empty /etc/fstab. weird, but I've seen it happen.
+            continue
+        roots.append(Root(mounts=mounts, swaps=swaps, name=name))
+
+    return roots
+
+
+class StorageDiscoveryConfig(object):
+
```
Comment (jkonecny12) [inline pyanaconda/storage/osinstall.py]: Remove this blank line.

File: pyanaconda/storage/osinstall.py [hunk 6]
```diff
@@ -0,0 +1,2197 @@
...
+        """
+        self.clear_part_type = ksdata.clearpart.type
+        self.clear_part_disks = ksdata.clearpart.drives[:]
+        self.clear_part_devices = ksdata.clearpart.devices[:]
+        self.initialize_disks = ksdata.clearpart.initAll
+        self.zero_mbr = ksdata.zerombr.zerombr
+
+
+class FSSet(object):
+
```
Comment (jkonecny12) [inline pyanaconda/storage/osinstall.py]: The same here.

File: pyanaconda/storage/osinstall.py [hunk 7]
```diff
@@ -0,0 +1,2197 @@
...
+        :type devices: iterable providing blivet.devices.StorageDevice instances holding
+                       a swap format
+
+        """
+
+        self._fstab_swaps = set(devices)
+
+
+class Root(object):
+
```
Comment (jkonecny12) [inline pyanaconda/storage/osinstall.py]: Remove this blank line please too.

File: pyanaconda/storage/osinstall.py [hunk 8]
```diff
@@ -0,0 +1,2197 @@
...
+        if not self.name and "/" in self.mounts:
+            self.name = self.mounts["/"].format.uuid
+
+    @property
+    def device(self):
+        return self.mounts.get("/")
+
+
+class BlkidTab(object):
+
```
Comment (jkonecny12) [inline pyanaconda/storage/osinstall.py]: And this too.

File: pyanaconda/storage/osinstall.py [hunk 9]
```diff
@@ -0,0 +1,2197 @@
...
+    def __getitem__(self, key):
+        return self.devices[key]
+
+    def get(self, key, default=None):
+        return self.devices.get(key, default)
+
+
+class CryptTab(object):
+
```
Comment (jkonecny12) [inline pyanaconda/storage/osinstall.py]: And here.

File: pyanaconda/storage/osinstall.py [hunk 10]
```diff
@@ -0,0 +1,2197 @@
...
+    def create_free_space_snapshot(self):
+        self._free_space_snapshot = self.get_free_space()
+
+        return self._free_space_snapshot
+
+    def get_free_space(self, disks=None, clear_part_type=None):  # pylint: disable=arguments-differ
+        """ Return a dict with free space info for each disk.
+
+             The dict values are 2-tuples: (disk_free, fs_free). fs_free is
```
Comment (jkonecny12) [inline pyanaconda/storage/osinstall.py]: Bad indentation.

File: tests/pyanaconda_tests/clearpart_test.py [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+            magic partitions
+            non-empty partition table
+        """
+        pass
+
+    def test_recursive_remove(self):
+        """
+            protected device at various points in stack
+        """
+        pass
```
Comment (jkonecny12) [inline tests/pyanaconda_tests/clearpart_test.py]: Why we have tests which are not testing anything?

**Pull request merged**